src/hotspot/share/classfile/protectionDomainCache.cpp

```diff
@@ -35,23 +35,24 @@
 #include "oops/oop.inline.hpp"
 #include "oops/weakHandle.inline.hpp"
 #include "runtime/atomic.hpp"
+#include "runtime/mutexLocker.hpp"
 #include "utilities/growableArray.hpp"
-#include "utilities/hashtable.inline.hpp"
+#include "utilities/resourceHash.hpp"
 
-unsigned int ProtectionDomainCacheTable::compute_hash(Handle protection_domain) {
-  // Identity hash can safepoint, so keep protection domain in a Handle.
-  return (unsigned int)(protection_domain->identity_hash());
+unsigned int ProtectionDomainCacheTable::compute_hash(const WeakHandle& protection_domain) {
+  return (unsigned int)(protection_domain.resolve()->identity_hash());
```

Comment:
And if it is a `WeakHandle` can't `resolve` now return NULL?

Reply:
compute_hash() is always called on a live WeakHandle so will never return null.

Reply:
I think you should add an assert that the `protection_domain.resolve()` never returns null, with a comment explaining why.

Reply:
I agree. In general if code is using WeakHandle's then I would expect to see null checks.

Reply:
I added the comment.